crates/matrix-sdk-sqlite/src/crypto_store.rs

```diff
@@ -1111,6 +1111,12 @@ impl CryptoStore for SqliteCryptoStore {
             .transpose()
     }
 
+    async fn delete_dehydrated_device_pickle_key(&self) -> Result<(), Self::Error> {
+        let conn = self.acquire().await?;
```

Comment:
Should I take a lock on `let _guard = self.save_changes_lock.lock().await;` here ?
In case of a race between that and save_changes??

Reply:
I don't think you need to do that since the transaction/connection will already act as a lock. The `save_changes_lock()` is necessary because we have a bunch of await points before we acquire the SQLite transaction.